src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaImplementationTestsBase.cs

```diff
@@ -0,0 +1,139 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Test.Cryptography;
+using Xunit;
+
+namespace System.Security.Cryptography.SLHDsa.Tests
+{
+    [ConditionalClass(typeof(SlhDsa), nameof(SlhDsa.IsSupported))]
+    public abstract class SlhDsaImplementationTestsBase : SlhDsaInstanceTestsBase
+    {
+        public static IEnumerable<object[]> NistSigVerTestVectorsData =>
+            from vector in SlhDsaTestData.NistSigVerTestVectors
```

Comment:
GH won't let me leave feedback in the big file, so I'll leave it here:

Consider adding a ToString on the NistSigVerTestVectors class; it should help with printing/identifying test cases.

Make a test fail one theory input, look at how it represents it.  Then add a ToString and fail it again, tweak until it feels useful.

https://github.com/dotnet/runtime/pull/114485/commits/4fba8bae4f43770c112bd12fb95193c96c1db811 for Krzysztof doing that in ML-DSA.

Reply:
> Consider adding a ToString on the NistSigVerTestVectors class

`SlhDsaSigVerTestVector` is a `record`. It already includes a `ToString` implementation that is (good enough, in my opinion).

Reply:
Is it not going to include a HUGE output from all the hex strings?

Reply:
Ah, yeah, it probably will.

Reply:
Ah, I see. Note that in ML-KEM I intentionally avoided using these as inputs to `Theory`. One of the feedback items I got for SHA-3 is to avoid having theories that are 1000s of cases. Regardless of what the `ToString` of the `record` does, it created explosively large `TestResults.xml` results. XUnit doesn't care about `ToString`. It cares about implementing `IXunitSerializable`.

So a `ToString` I think would be nice, but I would be careful about using Theory in combination of massive inputs, otherwise the resulting XML file is many megabytes.

Reply:
That's a fair point.  In both this and ML-DSA I believe theoryData.Count() to be "low", but theoryDatum.ToStringOrImplicitCaseName().Length to be "whoa, there!" big.  If my assumption is invalid then some other strategy is warranted.

Reply:
I like having the algo in the test parameters (and the test case id for uniqueness) for test failure reporting. I'll just update the theory input to only include those as members and look up the rest at runtime from a dictionary keyed by the test case id.

Reply:
I verified that `TestResults.xml` uses what `ToString` returns so I just overrode that.